src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1460,18 +1460,57 @@ public static void SetTestHook(string property, object value)
         }
     }
 
+    internal class HistoryStack<T>
```

Comment:
`BoundedStack<T>` below has a public doc comment. I expect we add a public comment here too with an explanation why we use it and what is the difference from `BoundedStack<T>`.